pkgs/by-name/so/somo/package.nix

```
@@ -0,0 +1,57 @@
+{
+  lib,
+  stdenv,
+  rustPlatform,
+  fetchFromGitHub,
+  installShellFiles,
+  versionCheckHook,
+  nix-update-script,
+}:
+
+rustPlatform.buildRustPackage (finalAttrs: {
+  pname = "somo";
+  version = "1.1.0";
+
+  src = fetchFromGitHub {
+    owner = "theopfr";
+    repo = "somo";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-HUTaBSy3FemAQH1aKZYTJnUWiq0bU/H6c5Gz3yamPiA=";
+  };
+
+  cargoHash = "sha256-e3NrEfbWz6h9q4TJnn8jnRmMJbeaEc4Yo3hFlaRLzzQ=";
+
+  nativeBuildInputs = [
+    installShellFiles
+  ];
+
+  postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+    installShellCompletion --cmd somo \
+      --bash <("$out/bin/somo" generate-completions bash) \
+      --zsh <("$out/bin/somo" generate-completions zsh) \
+      --fish <("$out/bin/somo" generate-completions fish)
+  '';
+
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+  doInstallCheck = true;
+  versionCheckProgram = "${placeholder "out"}/bin/somo";
+  versionCheckProgramArg = "--version";
+
+  passthru = {
+    updateScript = nix-update-script { };
+  };
+
+  meta = {
+    description = "Socket and port monitoring tool";
+    homepage = "https://github.com/theopfr/somo";
+    changelog = "https://github.com/theopfr/somo/blob/${finalAttrs.src.tag}/CHANGELOG.md";
```

```suggestion
    changelog = "https://github.com/theopfr/somo/blob/v${finalAttrs.version}/CHANGELOG.md";
```

Please don't use `finalAttrs.src` here since this would break when overriding the `src` attribute with something that doesn't have a `tag` (e.g. a local path).

---

Thank you for pointing out! I often forgot it :bow: